assets/src/edit-story/components/canvas/canvasUploadDropTarget.js

```diff
@@ -49,17 +54,70 @@ const PageAreaCover = styled(PageArea)`
 
 function CanvasUploadDropTarget({ children }) {
   const { uploadFile } = useUploader();
+  const {
+    state: { media, mediaType, searchTerm },
+    actions: { fetchMediaSuccess },
+  } = useMedia();
+  const {
+    actions: { updateElementById },
+  } = useStory();
   const insertElement = useInsertElement();
   const onDropHandler = useCallback(
-    (files) => {
-      files.forEach((file) => {
-        uploadFile(file).then((res) => {
-          const resource = getResourceFromUploadAPI(res);
-          insertElement(resource.type, { resource });
+    async (files) => {
```

Comment:
I think we should move this logic to the media provider, where it belongs.

Reply:
Still a WIP one :D I will take this note.